src/harness/harnessLanguageService.ts

```diff
@@ -333,6 +333,9 @@ module Harness.LanguageService {
         getDefinitionAtPosition(fileName: string, position: number): ts.DefinitionInfo[] {
             return unwrapJSONCallResult(this.shim.getDefinitionAtPosition(fileName, position));
         }
+        getTypeDefinitionAtPosition(fileName: string, position: number): ts.DefinitionInfo[]{

```

Comment:
One gets you the place where the "symbol" is defined, where the other takes you to the definition location of the "type".

it is a very useful feature when you want to know what is the shape of `value.property`, hovering gives you a type name, but to know the shape of the type you need to first jump to the declaration of `property`, then jump to the declaration of its type. it gets even harder with inferred types, cause you need to go to the definition of the function, or variable that produces the contextual type then go to the type from there.

editors like Eclipse and monaco have had this feature for long time.